### DISCUSSION TITLE
Possible to build deb file with checkinstall?

### DISCUSSION THREAD

**Discussion by mspengle** [abraunegg/onedrive#2764] [Q&A] [answered]
Is there a way to build a deb file with the checkinstall command? I tried and it fails. tia

**Comment by abraunegg** [comment, marked as answer]
Using `checkinstall` to build a `.deb` file is possible, but there are significant reasons why it may not be the best tool for the job. While `checkinstall` provides a convenient way to convert compiled source code into a Debian package, it comes with several notable issues and potential security concerns that you should be aware of. Here are the key points to consider:

1. **Removal from Debian Testing Suite**:
   `checkinstall` has been removed from the Debian testing suite due to migration policy violations and potential bugs. This removal indicates that the tool does not meet the quality and stability standards required for Debian's testing branch, which is a serious concern for its reliability and safety

2. **System Bricking Risk**:
   One of the significant issues with `checkinstall` is the risk of rendering your system non-bootable if the process is interrupted. Users have reported cases where interrupting `checkinstall` during its operation led to a "bricked" system, particularly on Ubuntu 22.04. This is a severe risk that can lead to substantial downtime and recovery efforts

3. **Root Privileges Requirement**:
   `checkinstall` requires root privileges for most of its operations, which increases the risk of accidental system damage. Although there are methods to run it without root, these workarounds are not always straightforward and may not be suitable for all scenarios

4. **Improper Package Creation**:
   Often, `checkinstall` may not correctly interpret the `make install` process, leading to incomplete or improperly configured packages. This can result in residual files remaining on the system even after the package is uninstalled, causing potential conflicts and clutter

5. **Filesystem Translation Issues**:
   The default behavior of `checkinstall` to perform filesystem translation (using the `--fstrans` option) is meant to prevent direct modifications to the system directories. However, if this feature is disabled or not properly handled, it can lead to system corruption

6. **Dependency Handling**:
   `checkinstall` does not manage dependencies as effectively as native packaging tools like `apt` or `yum`. This can leave unmanaged dependencies when installing software with `checkinstall`, causing conflicts or issues when other software is installed or updated

7. **Directory Creation Management**:
   Another critical issue is that `checkinstall` cannot manage the creation of necessary directories during the installation process. This limitation can result in failed installations when the software being installed depends on specific directories that are not present or created by the `checkinstall` process

Given these significant drawbacks, it's generally recommended to use more robust and reliable tools for creating Debian packages, such as `dpkg-deb` or using a proper Debian packaging workflow. These tools are better integrated with the system's package management and provide more reliable dependency handling and installation processes.


Your `checkinstall` process is most likely failing by points 4 and 7 above - this is nothing to fix or solve here. The recommendation here is do not use `checkinstall` to build your package.

### References
1. [Debian Wiki on CheckInstall](https://wiki.debian.org/CheckInstall)
2. [Debian Package Tracker for CheckInstall](https://tracker.debian.org/pkg/checkinstall)
3. [CheckInstall Manpage - Debian Manpages](https://manpages.debian.org/buster/checkinstall/checkinstall.8.en.html)
4. [Checkinstall can "brick" your Ubuntu 22.04 system - Reddit](https://www.reddit.com/r/Ubuntu/comments/x08ex6/psa_checkinstall_can_brick_your_ubuntu_2204_system/)
5. [Does using checkinstall instead of make install allow to easily remove dependencies? - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/660306/does-using-checkinstall-instead-of-make-install-allow-to-easily-remove-dependenc)
6. [Checkinstall ignores install=no; how to have it build, but NOT autoinstall a package? - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/53871/debian-checkinstall-ignores-install-no-how-to-have-it-build-but-not-autoinstal)
7. [CheckInstall Issue - GitHub Pull Request for onedrive](https://github.com/abraunegg/onedrive/pull/1107)
